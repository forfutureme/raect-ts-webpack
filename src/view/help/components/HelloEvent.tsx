/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-09 08:19:00
 * @Desc: Hello 事件例子
 */
import React from 'react'

interface props {
  click?: () => void
}

const HelloEvent: React.FC<props> = ({click}) => {
  return (
    <div>
      <h3>HelloEvent</h3>
      <button onClick={click}>点击我</button>
    </div>
  )
}

export default HelloEvent
