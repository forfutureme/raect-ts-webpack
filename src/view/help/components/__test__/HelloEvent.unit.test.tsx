/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-09 08:27:00
 * @Desc: HelloEvent 单元测试
 */
import renderer from 'react-test-renderer'
import {mount} from 'enzyme'
import React from 'react'

import HelloEvent from '../HelloEvent'

describe('HelloEvent组件', () => {
  it('测试组件渲染', () => {
    const json = renderer.create(<HelloEvent />).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('测试组件渲染-传了click方法', () => {
    const fn = () => {
      console.log(0)
    }
    const json = renderer.create(<HelloEvent click={fn} />).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('测试点击方法', () => {
    const fn = jest.fn()
    const component = mount(<HelloEvent click={fn} />)
    component.find('button').simulate('click')
    expect(fn).toBeCalled()
  })
})
