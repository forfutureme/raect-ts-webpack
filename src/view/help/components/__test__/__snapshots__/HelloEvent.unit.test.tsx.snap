// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloEvent组件 测试组件渲染 1`] = `
<div>
  <h3>
    HelloEvent
  </h3>
  <button>
    点击我
  </button>
</div>
`;

exports[`HelloEvent组件 测试组件渲染-传了click方法 1`] = `
<div>
  <h3>
    HelloEvent
  </h3>
  <button
    onClick={[Function]}
  >
    点击我
  </button>
</div>
`;
